client/js/app/stores/ExplorerStore.js

```diff
@@ -84,6 +101,87 @@ function _getDefaultFilterCoercionType(explorer, filter) {
 }
 
 /**
+ * Runs through the changeset and moves data around if necessary
+ * @param {Object} explorer   The explorer model that is being updated
+ * @param {Object} updates    The updated explorer model
+ * @return {Object}           The new set of updates
+ */
+function _prepareUpdates(explorer, updates) {
+  var newModel = _.assign({}, explorer, updates);
+
+  newModel = _removeEmailExtractionFields(explorer, newModel);
+  newModel = _migrateFunnelSteps(explorer, newModel);
+
+  return newModel;
+}
+
+/**
+ * If the query got changed to a funnel, move the step-specific parameters to a steps object.
+ * and vice versa if it got changed FROM a funnel
+ * @param {Object} explorer The explorer model that is being updated
+ * @param {Object} newModel The updated explorer model
+ * @return {Object}         The new set of updates
+ */
+function _migrateFunnelSteps(explorer, newModel) {
+  var sharedProperties = ['event_collection', 'time', 'timezone', 'filters']
+  if(newModel.query.analysis_type === 'funnel' && explorer.query.analysis_type !== 'funnel') {
+    // Changing TO funnels
+    var firstStep = _defaultStep();
+    firstStep.active = true;
+
+    _.each(sharedProperties, function (key) {
+      if(typeof(explorer.query[key]) !== 'undefined' && explorer.query[key] !== null) {
```

Comment:
I'd say that's a good rule if you are trying to replace everything, even simple things like the following with lodash

```js
var foo = foobar.foo;
// vs
var foo = _.get(foobar, 'foo');
```

But since you are using lodash in your project, and you're doing what feels a bit more "manual" than `_.isUndefined` or `_.isNull`, I don't see any reason to not use a simpler syntax that lodash provides.

Unless you plan on leaving lodash anytime soon.